### PR TITLE
feat(django): set `usr.id` tag by default if `request.user` is authenticated

### PR DESCRIPTION
**Pull request by avara1986**
## Description
Django: set `usr.id` tag by default if `request.user` is authenticated

## Checklist
- [ ] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

**Review by gnufede — APPROVED** [review]

File: ddtrace/contrib/trace_utils.py [hunk 1]
```diff
@@ -569,21 +569,21 @@ def set_user(tracer, user_id, name=None, email=None, scope=None, role=None, sess
...
         # Required unique identifier of the user
-        span.set_tag(user.ID, user_id)
+        span._set_str_tag(user.ID, user_id)
         if propagate:
             span.context.dd_user_id = user_id
 
         # All other fields are optional
         if name:
-            span.set_tag(user.NAME, name)
+            span._set_str_tag(user.NAME, name)
```
Comment (P403n1x87) [inline ddtrace/contrib/trace_utils.py:578]: These changes seem to be unrelated to the PR?

File: releasenotes/notes/django-set-usr-id-by-default-b99c0f3e6cf530a2.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Django: set `usr.id` tag by default if `request.user` is authenticated
```
Comment (P403n1x87) [inline releasenotes/notes/django-set-usr-id-by-default-b99c0f3e6cf530a2.yaml:4]: missing new line at the EOF

**Review by juanjux — APPROVED** [review]

**Review by gnufede — APPROVED** [review]
